R/teal_data_module.R

```diff
@@ -61,15 +61,71 @@
 #' @export
 teal_data_module <- function(ui, server, label = "data module", once = TRUE) {
   checkmate::assert_function(ui, args = "id", nargs = 1)
-  checkmate::assert(
-    checkmate::check_function(server, args = "id", nargs = 1),
-    # todo: allow for teal_data_module$server to have 'data' argument or break this in teal_transformer_module
-    checkmate::check_function(server, args = c("id", "data"), nargs = 2)
-  )
+  checkmate::check_function(server, args = "id", nargs = 1)
```

Comment:
Please make tests and fix failing ones.
```suggestion
  checkmate::assert_function(server, args = "id", nargs = 1)
```